docs/solidity-by-example.rst

```diff
@@ -746,46 +718,28 @@ creates the proper signature for the ``ReceiverPays`` example:
     // nonce can be any unique number to prevent replay attacks
     // contractAddress is used to prevent cross-contract replay attacks
     function signPayment(recipient, amount, nonce, contractAddress, callback) {
-        var hash = "0x" + ethereumjs.ABI.soliditySHA3(
+        var hash = "0x" + abi.soliditySHA3(
             ["address", "uint256", "uint256", "address"],
             [recipient, amount, nonce, contractAddress]
         ).toString("hex");
-        
-        web3.personal.sign(hash, web3.eth.defaultAccount, callback);
+
+        web3.eth.personal.sign(hash, web3.eth.defaultAccount, callback);
     }
 
 Recovering the Message Signer in Solidity
 -----------------------------------------
 
-In general, ECDSA signatures consist of two parameters, ``r`` and ``s``.
-Signatures in Ethereum include a third parameter called ``v``, that can be used
-to recover which account's private key was used to sign in the message,
-the transaction's sender. Solidity provides a built-in function
-`ecrecover <mathematical-and-cryptographic-functions>`_
-that accepts a message along with the ``r``, ``s`` and ``v`` parameters and
-returns the address that was used to sign the message.
+In general, ECDSA signatures consist of two parameters, ``r`` and ``s``. Signatures in Ethereum include a third parameter called ``v``, that you can use to recover which account's private key was used to sign the message, and the transaction's sender. Solidity provides a built-in function `ecrecover <mathematical-and-cryptographic-functions>`_ that accepts a message along with the ``r``, ``s`` and ``v`` parameters and returns the address that was used to sign the message.
```

Comment:
This sounds a little like you would be able to recover the private key which is of course not possible.  Also the transaction's sender can be derived from the public key, it is not something you could also recover using v.